aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -405,6 +463,19 @@ Tensor _embedding_bag_backward(const Tensor &grad, const Tensor &indices,
   checkScalarType("embedding_bag", offsets_arg, kLong);
   checkContiguous("embedding_bag", offsets_arg);
 
+  // To save compute, if we are going to go down the fast path case for the 'sum'
+  // mode, we skip calculating offset2bag, since it is not going to be used.
+  auto sparse_fast_path_sum = [&grad, &per_sample_weights, &mode, &sparse]() {
```

Comment:
Why does this need to be a lambda vs just computing a boolean and branching on it?  It's strictly more compile time and runtime to do it this way.